bin/reference_tools/update_bed.py

```diff
@@ -228,22 +262,22 @@ def write_ensembl_bed(
 ):
     """Print padded exons for protein coding transcripts"""
 
-    ensembl_tmp_gz_fp = out_dir / "tmp.gtf.gz"
-    ensembl_tmp_fp = out_dir / "tmp.gtf"
+    ensembl_tmp_path = out_dir / "tmp.gtf.gz"
 
     if not skip_download:
-        gtf_request = get_gtf_request(release)
-        download_gtf_cmd = ["wget", gtf_request, "-O", ensembl_tmp_gz_fp]
-        subprocess.run(download_gtf_cmd, check=True)
-        gunzip_cmd = ["gunzip", ensembl_tmp_gz_fp]
-        subprocess.run(gunzip_cmd, check=True)
+        gtf_url = get_gtf_url(release)
+        requests.get(gtf_url, stream=True)
```

Comment:
```suggestion
        gtf_url = get_gtf_url(release)
        response = requests.get(gtf_url, stream=True)
        response.raise_for_status() 
        
        with open(ensembl_tmp_path, "wb") as gtf_download_fh:
            LOG.info(f"Fetching {gtf_url}, might take a while!")
            gtf_download_fh.write(response.content)
```

You forgot to save the response to yr target outfile. Code above should work.

`response.raise_for_status()` will raise an HTTPerror if an error occurs when establishing the connection.

Reply:
OK, looks like I never actually tested the change 🫣  Tested it out now.

Seems for the `stream=True` to do a benefit in keeping down the RAM usage the response needs to be written in chunks. Added that as well.